File: uoj_12150_pole-position.cpp
// Implementation of python version There is something wrong with
// python and this site 100%
#include <cstdio>

int main()
{
  int n,c,p;
  int tmpPos;
  while(scanf("%d",&n) && n)
    {
      int StartingLineUp[1000] = {0};
      bool impossibol = false;

      for (int position = 0; position < n; position++)
	{
	  scanf("%d %d", &c,&p);
	  tmpPos = position + p;
	  if ((tmpPos > n-1) || (tmpPos < 0) || StartingLineUp[tmpPos])
	    {
	      impossibol = true;
	      continue;
	    }
	  StartingLineUp[tmpPos] = c;
	}

      if (impossibol)
	printf("-1\n");
      else
	{
	  for (int car = 0; car < n-1; car++)
	    printf("%d ", StartingLineUp[car]);
	  printf("%d\n", StartingLineUp[n-1]);
	}
	 
      
    }
  return 0;
  
}

File: uoj_12150_pole-position.py
# Seems pretty easy
# Got wrong answer tho

N = int(input())

cases = 2

while N:
    startingLineUp = [0]*(N+1)
    impossibol = False
    for position in range(1,N+1):
        C,P = [int(i) for i in input().split()]
        supposedStartingPos = position + P
        if supposedStartingPos > N \
           or supposedStartingPos < 0\
           or startingLineUp[supposedStartingPos]:
            impossibol = True # possibol :P
            continue
            # break  - can't break need to see all of the input
        startingLineUp[supposedStartingPos] = C

    if impossibol:
        print("-1")
    else:
        for car in range(1,N):
            print(f"{startingLineUp[car]} ")
        print(f"{startingLineUp[-1]}")

    N = int(input())
